src/scripts.js

```diff
@@ -0,0 +1,225 @@
+
+
+//object contains stored parameters for searches
+  customParameters : {
+    q: {val: null, string: ""},
+    from: {val: null, string: ""},
+    excludeDomains: {val: null, string: ""},
+    language: {val: null, string: ""},
+    page: {val: 1, string: ""},
+    sortBy: {val: null, string: ""}
+  },
+
+
+//getURL provides a url for API with specified parameters above
+  getURL: function(){
+    const customURL = `https://newsapi.org/v2/everything?${this.customParameters.q.string}${this.customParameters.from.string}${this.customParameters.excludeDomains.string}${this.customParameters.page.string}${this.customParameters.language.string}${this.customParameters.sortBy.string}&apiKey=9ed005ef4eb94baf913fce701c69972f`
+        return customURL
+  },
+
+  updateURL: function(parameter, update){
+    this.customParameters[parameter].val = update
+    if (parameter === "q"){
+      this.customParameters[parameter].string = `${parameter}=${update}`
+    }else{
+      this.customParameters[parameter].string = `&${parameter}=${update}`
+    }
+  },
+
+
+  blockList: []
+
+}
+
+
+//Object contains functions attached to buttons and page features
+const pageHandlers = {
+
+  changeCountry: function(){
+    const countryButtonElement = document.querySelector(".country-button")
+      countryButtonElement.addEventListener("click", event => {
+        paginationElement.style.display = "none";
+        countryButtonElement.textContent === "Top UK Stories"
+          ? ( countryButtonElement.textContent = "Top US Stories",
+              fetchNews(apiRequests.ukTop20))
+          : (countryButtonElement.textContent = "Top UK Stories",
+              fetchNews(apiRequests.usTop20))
+      })
+    },
+
+  search: function(event){
+      event.preventDefault();
+      if (searchTextElement.value === ""){
+        return
+      }
+      const dateRange = formatDate(dateRangeElement.value)
+      paginationElement.style.display = "inline"
+      apiRequests.updateURL("q", searchTextElement.value)
+      apiRequests.updateURL("from", dateRange)
+      fetchNews(apiRequests.getURL())
+      searchTextElement.value = ""
+      pageHandlers.checkPage()
+    },
+
+
+  checkPage: function(){
+    apiRequests.customParameters.page.val === 1
+      ? prevPageElement.disabled = true
+      : prevPageElement.disabled = false
+  },
+
+
+  paginationControl: function(event){
+    let currentPage = apiRequests.customParameters.page.val
+    event.target.value === "next"
+      ? currentPage ++
+      : currentPage --
+    apiRequests.updateURL("page", currentPage)
+    fetchNews(apiRequests.getURL())
+    pageHandlers.checkPage()
+
+
+},
+
+  updateAdvancedOptions: event => {
+    event.preventDefault()
+    apiRequests.updateURL("language", languageSelector.value)
+    apiRequests.updateURL("sortBy", sortBySelector.value )
+    pageHandlers.search(event)
+},
+
+   openCloseAdvanced: event => {
+    advancedSearchMenu.style.display === "none"
```

Comment:
This could rewritten slightly cleaner as 
`advancedSearchMenu.style.display = advancedSearchMenu.style.display === "none ?  = "block" : "none"`